Lib/test/test_importlib/test_namespace_pkgs.py

```diff
@@ -318,6 +318,16 @@ def test_module_before_namespace_package(self):
         self.assertEqual(a_test.attr, 'in module')
 
 
+class NamespaceSubpackageSameName(NamespacePackageTest):
+    paths = ['']
+
+    def test_namespace_subpackage_shares_name_with_directory(self):
+        submodule_path = 'project4.foo'
+        msg = 'project4 must be imported before finding project4.foo.'
+        with self.assertRaisesRegex(ModuleNotFoundError, msg):
+            importlib.machinery.PathFinder.find_spec(submodule_path)
```

Comment:
Being so specific on the exception message is very brittle, so we tend to avoid it. Instead, test any of the data attached to the exception as that's something that code might work with.

```suggestion
        with self.assertRaisesRegex(ModuleNotFoundError) as cm:
            importlib.machinery.PathFinder.find_spec(submodule_path)
            
        self.assertEqual(cm.excepion.name, 'project4')
```